tests/src/test/scala/actionContainers/DotNetActionContainerTests.scala

```diff
@@ -0,0 +1,227 @@
+      functionb64,
+      main = "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.Echo::Main"
+    )
+  }
+
+  override val testUnicode = {
+    TestConfig(
+      functionb64,
+      main = "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.Unicode::Main"
+    )
+  }
+
+  override val testInitCannotBeCalledMoreThanOnce = testEchoNoWrite
+
+  override val testEntryPointOtherThanMain = testEchoNoWrite
+
+  override val testLargeInput = testEchoNoWrite
+
+  it should "fail to initialize with bad archive" in {
+    val (out, err) = withActionContainer() { c =>
+      val brokenArchive = ("NOTAVALIDZIPFILE")
+
+      val (initCode, initRes) = c.init(initPayload(brokenArchive, "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.Invalid::Main"))
+      initCode should not be (200)
+
+      initRes shouldBe defined
+
+      initRes should {
+          be(Some(JsObject("error" -> JsString("Unable to decompress package."))))
+      }
+    }
+  }
+
+  it should "return some error on action error" in {
+    val (out, err) = withActionContainer() { c =>
+      val (initCode, _) = c.init(initPayload(functionb64, "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.Exception::Main"))
+      initCode should be(200)
+
+      val (runCode, runRes) = c.run(runPayload(JsObject.empty))
+      runCode should not be (200)
+
+      runRes shouldBe defined
+      runRes.get.fields.get("error") shouldBe defined
+    }
+
+    checkStreams(out, err, {
+      case (o, e) =>
+        (o + e).toLowerCase should include("exception")
+    })
+  }
+
+  it should "support application errors" in {
+    val (out, err) = withActionContainer() { c =>
+      val (initCode, _) = c.init(initPayload(functionb64, "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.Error::Main"))
+      initCode should be(200)
+
+      val (runCode, runRes) = c.run(runPayload(JsObject.empty))
+      runCode should be(200)
+
+      runRes shouldBe defined
+      runRes.get.fields.get("error") shouldBe defined
+    }
+
+    checkStreams(out, err, {
+      case (o, e) =>
+        o shouldBe empty
+        e shouldBe empty
+    })
+  }
+
+  it should "fails on invalid assembly reference" in {
+    val (out, err) = withActionContainer() { c =>
+      val (initCode, initRes) = c.init(initPayload(functionb64, "Apache.OpenWhisk.Tests.Dotnet.DoesntExist::Apache.OpenWhisk.Tests.Dotnet.Echo::Main"))
+      initCode should be(502)
+
+      initRes shouldBe defined
+
+      initRes should {
+          be(Some(JsObject("error" -> JsString("Unable to locate requested assembly (\"Apache.OpenWhisk.Tests.Dotnet.DoesntExist.dll\")."))))
+      }
+    }
+  }
+
+  it should "fails on invalid type reference" in {
+    val (out, err) = withActionContainer() { c =>
+      val (initCode, initRes) = c.init(initPayload(functionb64, "Apache.OpenWhisk.Tests.Dotnet::Apache.OpenWhisk.Tests.Dotnet.FakeType::Main"))
+      initCode should be(502)
+
+      initRes shouldBe defined
```

Comment:
`       initRes shouldBe defined ` not necessary next statement checks result is Some...